packages/colorpicker/lib/src/components/rgb_slider_group.dart

```diff
@@ -0,0 +1,287 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:paintroid/ui/theme/data/custom_colors.dart';
+import 'package:logging/logging.dart';
+import 'package:colorpicker/src/utils/upper_case_text_formatter.dart';
+import 'package:colorpicker/src/utils/hex_input_formatter.dart';
+
+final log = Logger('RgbSliderGroup');
+
+class RgbSliderGroup extends ConsumerStatefulWidget {
+  final Color initialColor;
+  final ValueChanged<Color> onColorChanged;
+
+  const RgbSliderGroup({
+    Key? key,
+    required this.initialColor,
+    required this.onColorChanged,
+  }) : super(key: key);
+
+  @override
+  ConsumerState<RgbSliderGroup> createState() => _SliderColorState();
+}
+
+class _SliderColorState extends ConsumerState<RgbSliderGroup> {
+  late int _red;
+  late int _green;
+  late int _blue;
+  late int _alphaValue;
+  late TextEditingController _hexController;
+
+  int _getAlpha(Color color) => (color.a * 255).round();
+
+  int _getRed(Color color) => (color.r * 255).round();
+
+  int _getGreen(Color color) => (color.g * 255).round();
+
+  int _getBlue(Color color) => (color.b * 255).round();
+
+  @override
+  void initState() {
+    super.initState();
+    _initializeStateFromColor(widget.initialColor);
+  }
+
+  @override
+  void didUpdateWidget(RgbSliderGroup oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.initialColor != oldWidget.initialColor) {
+      _initializeStateFromColor(widget.initialColor);
+    }
+  }
+
+  void _initializeStateFromColor(Color color) {
+    _red = _getRed(color);
+    _green = _getGreen(color);
+    _blue = _getBlue(color);
+    _alphaValue = _getAlpha(color);
+    _hexController = TextEditingController(text: _colorToHex(color));
```

Comment:
Please use _hexController = TextEditingController(...) in the initState once
Then after that update the text with _hexController.text =